joblib/test/test_numpy_pickle.py

```diff
@@ -178,28 +195,48 @@ def test_numpy_persistence():
             for item in obj_:
                 nose.tools.assert_true(isinstance(item, np.ndarray))
             # And finally, check that all the values are equal.
-            nose.tools.assert_true(np.all(np.array(obj) ==
-                                                np.array(obj_)))
+            np.testing.assert_array_equal(np.array(obj), np.array(obj_))
 
         # Now test with array subclasses
-        for obj in (
-                    np.matrix(np.zeros(10)),
-                    np.core.multiarray._reconstruct(np.memmap, (), np.float)
-                   ):
+        for obj in (np.matrix(np.zeros(10)),
+                    np.core.multiarray._reconstruct(np.memmap, (), np.float)):
             this_filename = filename + str(random.randint(0, 1000))
             filenames = numpy_pickle.dump(obj, this_filename,
-                                          compress=compress,
-                                          cache_size=cache_size)
+                                          compress=compress)
+            # All is cached in one file
+            nose.tools.assert_equal(len(filenames), 1)
+
             obj_ = numpy_pickle.load(this_filename)
-            if (type(obj) is not np.memmap
-                        and hasattr(obj, '__array_prepare__')):
+            if (type(obj) is not np.memmap and
+                    hasattr(obj, '__array_prepare__')):
                 # We don't reconstruct memmaps
                 nose.tools.assert_true(isinstance(obj_, type(obj)))
+                np.testing.assert_array_equal(obj_, obj)
 
-    # Finally smoke test the warning in case of compress + mmap_mode
-    this_filename = filename + str(random.randint(0, 1000))
-    numpy_pickle.dump(a, this_filename, compress=1)
-    numpy_pickle.load(this_filename, mmap_mode='r')
+        # Test with an object containing multiple numpy arrays
+        obj = ComplexTestObject()
+        filenames = numpy_pickle.dump(obj, this_filename,
+                                      compress=compress)
+        # All is cached in one file
+        nose.tools.assert_equal(len(filenames), 1)
+
+        obj_loaded = numpy_pickle.load(this_filename)
+        nose.tools.assert_true(isinstance(obj_loaded, type(obj)))
+        np.testing.assert_array_equal(obj_loaded.array_float, obj.array_float)
+        np.testing.assert_array_equal(obj_loaded.array_int, obj.array_int)
+        np.testing.assert_array_equal(obj_loaded.array_obj, obj.array_obj)
+
+
+@with_numpy
+def test_numpy_persistence_bufferred_array_compression():
+    big_array = np.ones((_IO_BUFFER_SIZE + 100), dtype=np.uint8)
+    small_array = np.ones((100), dtype=np.uint8)
```

Comment:
I would just keep the big array, I reckon we have plenty of other tests already for small arrays.